src/worker.cc

```diff
@@ -153,6 +153,13 @@ rapidjson::Document from_string(const std::string& json, const valhalla_exceptio
   return d;
 }
 
+// fuction to add warnings to api object
+void add_warning(valhalla::Api& api, std::string warning_text){
+  
+  auto* warning = api.mutable_info()->mutable_warnings()->Add();
+  warning->set_description(warning_text);
+}
```

Comment:
good impuls! as long as we're just using it in this one function, we can also make it a lambda inside the function and you can capture the `api` pointer from the outer scope.